internal/pkg/endpoint/endpoint.go

```diff
@@ -12,53 +12,80 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  *******************************************************************************/
+
 package endpoint
 
 import (
+	"context"
 	"fmt"
 	"os"
+	"sync"
 	"time"
 
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/interfaces"
 	"github.com/edgexfoundry/go-mod-registry/registry"
 )
 
-type Endpoint struct {
+type endpoint struct {
+	ctx            context.Context
+	wg             *sync.WaitGroup
 	RegistryClient *registry.Client
+	serviceKey     string // The key of the service as found in the service registry (e.g. Consul)
+	path           string // The path to the service's endpoint following port number in the URL
+	interval       int    // The interval in milliseconds governing how often the client polls to keep the endpoint current
+
 }
 
-func (e Endpoint) Monitor(params types.EndpointParams) chan string {
-	ch := make(chan string, 1)
-	ch <- e.fetch(params)
+func New(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	registryClient *registry.Client,
+	serviceKey string,
+	path string,
+	interval int) *endpoint {
+
+	return &endpoint{
+		ctx:            ctx,
+		wg:             wg,
+		RegistryClient: registryClient,
+		serviceKey:     serviceKey,
+		path:           path,
+		interval:       interval,
+	}
+}
+
+func (e endpoint) Monitor() chan interfaces.URLStream {
+	ch := make(chan interfaces.URLStream, 1)
+	url, err := e.buildURL()
+	if err != nil {
+		_, _ = fmt.Fprintln(os.Stdout, err.Error())
+	}
+	ch <- interfaces.URLStream(url)
+
+	e.wg.Add(1)
 	go func() {
+		defer e.wg.Done()
 		for {
-			url, err := e.buildURL(params)
+			select {
+			case <-e.ctx.Done():
+				break
+			}
```

Comment:
This last iteration, you moved this.  

I'm after a more comprehensive solution that uses time.NewTimer() and integrates the case <-timer.C in the select.